packages/core/src/connectors/browser.ts

```diff
@@ -0,0 +1,156 @@
+import { AccountInterface } from 'starknet'
+import { Connector } from './base'
+import { IStarknetWindowObject } from 'get-starknet'
+import {
+  ConnectorNotConnectedError,
+  ConnectorNotFoundError,
+  UserNotConnectedError,
+  UserRejectedRequestError,
+} from '../errors'
+
+export interface BrowserConnectorOptions {
+  id: string
+}
+
+export class BrowserConnector extends Connector<BrowserConnectorOptions> {
+  private _wallet?: IStarknetWindowObject
+
+  constructor({ options }: { options: BrowserConnectorOptions }) {
+    super({ options })
+  }
+
+  available(): boolean {
+    this.ensureWallet()
+    return this._wallet !== undefined
+  }
+
+  async ready(): Promise<boolean> {
+    this.ensureWallet()
+
+    if (!this._wallet) return false
+    return await this._wallet.isPreauthorized()
+  }
+
+  async connect(): Promise<AccountInterface> {
+    this.ensureWallet()
+
+    if (!this._wallet) {
+      throw new ConnectorNotFoundError()
+    }
+
+    try {
+      await this._wallet.enable()
+    } catch {
+      // NOTE: Argent v3.0.0 swallows the `.enable` call on reject, so this won't get hit.
+      throw new UserRejectedRequestError()
+    }
+
+    if (!this._wallet.isConnected) {
+      // NOTE: Argent v3.0.0 swallows the `.enable` call on reject, so this won't get hit.
+      throw new UserRejectedRequestError()
+    }
+
+    return this._wallet.account
+  }
+
+  async disconnect(): Promise<void> {
+    this.ensureWallet()
+
+    if (!this.available()) {
+      throw new ConnectorNotFoundError()
+    }
+
+    if (!this._wallet?.isConnected) {
+      throw new UserNotConnectedError()
+    }
+  }
+
+  async account(): Promise<AccountInterface> {
+    this.ensureWallet()
+
+    if (!this._wallet) {
+      throw new ConnectorNotConnectedError()
+    }
+
+    // FIXME This type is wrong. account can be null if user didn't connect wallet
+    return this._wallet.account
+  }
+
+  id(): string {
+    this.ensureWallet()
+    if (!this._wallet) {
+      throw new ConnectorNotConnectedError()
+    }
+    return this._wallet.id
+  }
+
+  name(): string {
+    this.ensureWallet()
+    if (!this._wallet) {
+      throw new ConnectorNotConnectedError()
+    }
+    return this._wallet.name
+  }
+
+  private ensureWallet() {
+    const installed = getInstalledWallets()
+    const wallet = installed[this.options.id]
+    if (wallet) {
+      this._wallet = wallet
+    }
+  }
+}
+
+// The code below comes from get-starknet and it's just to show what
+// we need from that library
+function getInstalledWallets(): Record<string, IStarknetWindowObject> {
```

Comment:
why not just use `getInstalledWallets` from `get-starknet`? it's core only (no UI)

Reply:
plus, this `getInstalledWallets` version doesn't prioritize pre-authorized wallets before non-authorized wallets, and for same-authorization state wallets - it's hardcodely puts the argent wallet first. these 2 issues are solved by the core-only `getInstalledWallets` method on `get-starknet`

Reply:
I'm having an issue with the package, but the main issue is that by detecting if the wallet is pre-authorized the function becomes async which doesn't work for this API. The idea of this PR is to figure out the minimum amount of code we need for the library, which then can be exported by `get-starknet`.

This library doesn't enforce any ordering, it's the developers that decide how to present wallets to their users. If you provide a function that simply returns a (shuffled) list of wallet ids I can re-export it from starknet react.

Reply:
```
export const shuffle = <T extends any[]>(arr: T): T => {
    for (let i = arr.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [arr[i], arr[j]] = [arr[j], arr[i]];
    }
    return arr;
};
```

Reply:
> I'm having an issue with the package

but the most problematic part from my POV is this part; could u pls list the missing features/APIs (besides of separating the `showModal` part to a separate lib)?

Reply:
I changed the default code to show a shuffled list of wallets.